pkg/gauges/contracts/AuthorizerAdaptor.sol

```diff
@@ -0,0 +1,105 @@
+ * and the Balancer Authorizer such that the Authorizer may grant/revoke admin powers to unlimited addresses. 
+ *
+ * The permissions the Authorizer can grant are granular such they may be global or specific to a particular contract
+ * @dev When calculating the actionId to call a function on a target contract, it must be calculated as if it were 
+ * to be called on this adaptor. This can be done by passing the function selector to the `getActionId` function.
+ */
+contract AuthorizerAdaptor is IAuthentication {
+    using Address for address;
+
+    bytes32 private immutable _actionIdDisambiguator;
+    IVault private immutable _vault;
+
+    constructor(IVault vault) {
+        // AuthorizerAdaptor is a singleton, so it simply uses its own address to disambiguate action identifiers
+        _actionIdDisambiguator = bytes32(uint256(address(this)));
+        _vault = vault;
+    }
+
+    /**
+     * @notice Returns the Balancer Vault
+     */
+    function getVault() public view returns (IVault) {
+        return _vault;
+    }
+
+    /**
+     * @notice Returns the Authorizer
+     */
+    function getAuthorizer() external view returns (IAuthorizer) {
+        return _getAuthorizer();
+    }
+
+    function _getAuthorizer() internal view returns (IAuthorizer) {
+        return getVault().getAuthorizer();
+    }
+
+    /**
+     * @dev As all contracts managed by this adaptor share the same action ID disambiguator
+     *      it's then important to be more granular over `where` an `account` is authorized to act.
+     */
+    function _canPerform(bytes32 actionId, address account, address where) internal view returns (bool) {
+        return _getAuthorizer().canPerform(actionId, account, where);
+    }
+
+    /**
+     * @notice Returns the action ID associated with calling a given function through this adaptor
+     * @dev The contracts managed by this adaptor do not have action ID disambiguators we use the adaptor's globally
+     * @param selector - The 4 byte selector of the function to be called using `performAction`
+     * @return The associated action ID
+     */
+    function getActionId(bytes4 selector) public view override returns (bytes32) {
+        // Each external function is dynamically assigned an action identifier as the hash of the disambiguator and the
+        // function selector. Disambiguation is necessary to avoid potential collisions in the function selectors of
+        // multiple contracts.
+        return keccak256(abi.encodePacked(_actionIdDisambiguator, selector));
+    }
+
+    /**
+     * @notice Performs an arbitrary function call on a target contract, provided the caller is authorized to do so.
+     * @param target - Address of the contract to be called
+     * @param data - Calldata to be sent to the target contract
+     * @return The bytes encoded return value from the performed function call
+     */
+    function performAction(address target, bytes calldata data) external returns (bytes memory) {
```

Comment:
I'd make this `nonReentrant` and `payable`, forwarding `msg.value` to `target`.

Reply:
Agreed on `nonReentrant`. I don't see much value in making this payable as I can't think of any admin functions which require ETH payments but I can add for generality's sake.